src/js/components/MaskedInput/MaskedInput.js

```diff
@@ -200,20 +200,20 @@ class MaskedInput extends Component {
 
   // This could be due to a paste or as the user is typing.
   onChange = event => {
-    const { onChange, mask } = this.props;
+    const { onChange, mask, ...props } = this.props;
     const {
       target: { value },
     } = event;
     // Align with the mask.
     const valueParts = parseValue(mask, value);
     const nextValue = valueParts.map(part => part.part).join('');
     if (onChange) {
-      onChange({ target: { ...event.target, value: nextValue } });
+      onChange({ target: { ...props, value: nextValue } });
```

Comment:
We seen to be losing the original event target. I think the key isn't to pass the component props but to make it look more like other events. In TextInput, the event contains the name because we pass the entire event object. I think what is needed here is more like: `onChange({ ...event, target: { ...event.target, value: nextValue } })`

Reply:
Right. I agree with the intent to make it look like other events. 

I attempted that approach, however when testing a MaskInput for phone number entry like shown below, we are losing any of the props the user was mentioning (e.g. `event.target.name` is `undefined`):
```
<MaskedInput
  mask={[
    { fixed: "(" },
    {
      length: 3,
      regexp: /^[0-9]{1,3}$/,
      placeholder: "xxx"
    },
    { fixed: ")" },
    { fixed: " " },
    {
      length: 3,
      regexp: /^[0-9]{1,3}$/,
      placeholder: "xxx"
    },
    { fixed: "-" },
    {
      length: 4,
      regexp: /^[0-9]{1,4}$/,
      placeholder: "xxxx"
    }
 ]}
 value={phone}
 name="Test MaskInput Phone  "
 onChange={event => {
   console.log("Event phone:", event)
   console.log("Event phone:", event.target)
   console.log("Event phone:", event.target.value)
   console.log("Event phone:", event.target.name)
  }}
/>
```

However, I believe:
```
onChange({ ...event, target: { ...props, value: nextValue } })
```
gives us both the event as you mention, plus the component props are added to the event like the user is expecting. Thoughts on this approach?

Reply:
I don't think we pass the component props through on any other events. Can you elaborate on why that is needed here?

Reply:
I don't have a use case particular to me. I was trying to solve for this posted issue: https://github.com/grommet/grommet/issues/3137. 

I read his post as he would like/expect the MaskedInput behavior to have similar target properties to that of TextInput. I am not sure what he is doing with them down stream. However he is using them for a TextInput, he would then be able to swap in a MaskedInput for his TextInput and maintain his downstream uses.

Again, I was just trying to meet the original request, so can't elaborate much beyond that.

Reply:
Ah, got it, thanks. In that case, I think the key requirement is to have the `name` available, which should be coming from the event target DOM element. The problem is, the way we are trying to extend the event to update the value isn't going to work. We will have to come up with another mechanism. :(

Reply:
What the MaskedInput is doing is both enforcing certain characters but also inserting separators. We both want to remove characters the user type and add characters appropriate to the caret and the mask. I'm wondering if we should cycle this through the DOM somehow? What I am thinking is that when the onChange happens from the underlying `<input>` element, if the `parseValue()` output doesn't equal the value in the event, then we hold off on propagating the event, re-render with the input now containing the parseValue() output, and then trigger an `change` event, which will result in the event value equalling the parseValue() output and we can deliver the real event. There might be holes in this, but I thought I'd write it down as something to consider. Are you willing to take a stab at this? It's a bit more complicated than the original change.

Reply:
I can take a look at this approach.